### PR TITLE
Fixes to MMC tendency forcing

### PR DESCRIPTION
**Pull request by moprak-nrel**
## Summary

This PR tracks fixes to MMC tendency forcing. The idea behind tendency forcing is to apply a time/height varying body force based on an input file, which can be used either with outputs from `ABLMesoForcingMom` or just a general list of time/z varying body forces in `u, v, theta` and a time varying surface-flux. 

## Pull request type

Please check the type of change introduced:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [ ] new unit-test(s)
- [ ] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [x] on CPU 
- the regression tests
  - [ ] on GPU 
  - [x] on CPU 

## Additional background

Issue Number: #1058

### CODE REVIEW

**Comment by moprak-nrel** [comment]
Note that this is still in draft stage. I'm still in the process of tying some loose ends with the temperature flux, and need to run more tests to ensure this works with refinement zones. Docs and regtests are incoming too.

**Comment by hgopalan** [comment]
Thank you for taking up this task. How are we going to plan the work flow for the MMC coupling since there are several ways of doing it: 

1. Drive using WRF data boundary conditions - We will need a temperature perturbations in upstream to generate flow structures. We can use the method integrated into ERF. 
2. Drive using surface conditions from WRF - We would need heterogeneous roughness lengths, temperature/heat flux for this. 
3.  Drive using WRF data as forcing 
4. Drive using met-mast profile as forcing. 

Do you want to keep the different driving mechanisms separate or have a single unified framework for all of them?

**Comment by moprak-nrel** [comment]
We might want to move this discussion to a relevant issue, but some quick responses to your questions below:

> Thank you for taking up this task. How are we going to plan the work flow for the MMC coupling since there are several ways of doing it:
> 
>     1. Drive using WRF data boundary conditions - We will need a temperature perturbations in upstream to generate flow structures. We can use the method integrated into ERF.

I'm not entirely sure what you mean here, are you talking about an inflow boundary condition with temperature perturbations? Or is this on the wall?

>     2. Drive using surface conditions from WRF - We would need heterogeneous roughness lengths, temperature/heat flux for this.
I thought #1133 was for addressing some of these?

>     3. Drive using WRF data as forcing
Not sure what this means either, are we talking about time varying mean profiles? 3D flow-fields? boundary conditions? Some combination of all of those?

>     4. Drive using met-mast profile as forcing.
Is this a 1-D profile of velocities/temperatures? The current `ABLMesoMomForcing` should work right?

 
> Do you want to keep the different driving mechanisms separate or have a single unified framework for all of them?

It's hard to say without understanding all the driving mechanisms.

**Comment by lawrenceccheung** [comment]
Just to provide some context here, this PR is meant to help address https://github.com/Exawind/amr-wind/issues/1058.  This method is meant to provide the "internal forcing" MMC coupling approach, where the horizontally averaged profiles are driven to match given 1D profiles in z and in time.  In particular, this should help provide internal body forces for the wind farm runs, when the precursor runs are complete and `ABLMesoMomForcing` and `ABLMesoTempForcing` are turned off (@ewquon please correct anything I'm saying here).

The source of the forcing data in this case would be from the precursor run, and the source of the profile data for the precursor run could be from either WRF data, lidar profilers, or met-mast data, depending on what's available.  This method _won't_ provide any help for the "boundary forcing" approach, where we'd take inflow data from WRF/ERF simulations and drive AMR-Wind with that.  After we get this settled, we might have a meeting with @ewquon, @mchurchf, and any other interested parties to see where to take things next.

Lawrence

File: amr-wind/equation_systems/icns/source_terms/ABLMesoForcingMom.cpp [hunk 1]
```diff
@@ -53,29 +57,21 @@ void ABLMesoForcingMom::mean_velocity_init(const ABLMesoscaleInput& ncfile)
...
-    m_vAvg_vals.resize(vavg.ncell_line());
-
-    m_meso_avg_error.resize(vavg.ncell_line());
-
-    m_error_meso_avg_U.resize(vavg.ncell_line());
-    m_error_meso_avg_V.resize(vavg.ncell_line());
-
-    m_err_U.resize(vavg.ncell_line());
-    m_err_V.resize(vavg.ncell_line());
+    m_velAvg_ht.resize(m_nht);
```
Comment (marchdf) [inline amr-wind/equation_systems/icns/source_terms/ABLMesoForcingMom.cpp:70]: silly comment but I will be taking care of this at some point. Can you make this a snake case variable name?

**Review by marchdf — APPROVED** [review]

**Comment by marchdf** [comment]
Can you add some documentation about this? Or maybe it's already there? And also a reg test (unless that is also covered)?

**Comment by moprak-nrel** [comment]
Thanks for reviewing, doc changes and reg tests are incoming shortly, wanted to get started on the code review process.

**Comment by moprak-nrel** [comment]
@ewquon Would like to get your eyes on this PR if you get a chance too. For some reason I couldn't add you as a reviewer.

**Comment by moprak-nrel** [comment]
Documenting of some of my tests:

[Here](https://github.com/user-attachments/files/16949748/norm_diffs.pdf) are normed differences of statistics of velocity/theta, the three lines are:

1. `PR`: This PR vs the main branch using MMC forcing
2. `tendency`: Tendency forcing vs MMC forcing
3. `tendency interpolated`: Tendency forcing vs MMC forcing, but the input file is at a coarser time/space resolution

As can be seen, 1 and 2 match exactly, and there is only a small interpolation error in 3.

**Comment by moprak-nrel** [comment]
Documentation has no changes, this is already present [here](https://exawind.github.io/amr-wind/user/inputs_ABL_meso_forcing.html).

**Pull request merged**